src/buildbot.py

```diff
@@ -38,7 +39,7 @@ def Fail(warn_only=False):
   """Mark one step as failing, but keep going."""
   sys.stdout.flush()
   if warn_only:
-    sys.stdout.write('\n@@@STEP_WARNINGS@@@\n')
+    Warn()
```

Comment:
Oh, I didn't finish. The other option might be to keep the boolean flags, and put the printing together in one function with a boolean switch for the type.